client/rpc-api/src/system/mod.rs

```diff
@@ -108,4 +108,27 @@ pub trait SystemApi<Hash, Number> {
 	/// known block.
 	#[rpc(name = "system_syncState", returns = "SyncState<Number>")]
 	fn system_sync_state(&self) -> Receiver<SyncState<Number>>;
+
+	/// Sets the logging filter, adding the supplied directives to the defaults.
+	///
+	/// The syntax is identical to the CLI `<target>=<level>`:
+	///
+	/// `sync=debug,state=trace`
+	///
+	/// The default filters, including those provided to the CLI will be merged
+	/// with the directives given.
+	#[rpc(name = "system_setLogFilter", returns = "()")]
+	fn system_set_log_filter(&self, directives: String)
```

Comment:
Isn't this more "add_log_filter"?

Reply:
The distinction here is about whether we set *all* directives or if we keep the default ones and add a few more. You are correct that this is then technically more of an "add" than a "set" operation, but if we assume that most users will not mess with the default directives (or even know what they are) then it becomes more natural to consider this a "set".

Reply:
it adds the directives given, but if you call it again, it doesn't cumulatively add them to the directives given in previous call(s), only adds the directives of this call to the default directives: substrate defaults + CLI `-l`  directives. From this POV maybe set is better than add

Reply:
maybe `setLogFilterUsingDefaults` or similar?